src/workerd/api/kv.c++

```diff
@@ -26,6 +26,8 @@ static void validateKeyName(kj::StringPtr name) {
   JSG_REQUIRE(name != "", TypeError, "Key name cannot be empty.");
   JSG_REQUIRE(name != ".", TypeError, "\".\" is not allowed as a key name.");
   JSG_REQUIRE(name != "..", TypeError, "\"..\" is not allowed as a key name.");
+  JSG_REQUIRE(name.size() <= 512, TypeError, "UTF-8 encoded length of ", name.size(),
```

Comment:
Nit: For hard coded limits like this, I'd much prefer a constant defined. Non-blocking tho.

More importantly, some comment here as to why the limit is 512 would be helpful for future generations.